lib/reek/smell_warning.rb

```diff
@@ -15,22 +15,21 @@ class SmellWarning
     extend Forwardable
 
     # @public
-    attr_reader :context, :lines, :message, :parameters, :smell_detector, :source
-    def_delegators :smell_detector, :smell_type
+    attr_reader :context, :lines, :message, :parameters, :smell_type, :source
 
     # @note When using Reek's public API, you should not create SmellWarning
     #   objects yourself. This is why the initializer is not part of the
     #   public API.
     #
     # @quality :reek:LongParameterList { max_params: 6 }
-    def initialize(smell_detector, context: '', lines:, message:,
+    def initialize(smell_type, context: '', lines:, message:,
```

Comment:
Let's use this opportunity to add proper parameter documentation :)

Reply:
Done!